### DISCUSSION TITLE
Dev strategy for incremental calculations

### DISCUSSION THREAD

**Discussion by njames879** [DaveSkender/Stock.Indicators#1075] [Dev notes]
Hi, 

I’m starting out on a project to develop a database of the FTSE 100 stocks with historical OHLC values plus the daily values for some common indicators,  and of course, the Skender stock indicators are front and centre.

so, I’m trying to figure out the coding strategy.   I’ve got the historical market data, and I can see how to use the package to calculate and store the indicator values.   

However, I will need to update the OHLC values on a daily basis, and here’s my question.  As far as I can see, the library generates the indicator value for each entry in the data set. When I do a daily update, against a library of say 500 historical daily values, I already have the indicator values for the 500 historical values,  so it seems like a huge waste of processing power to regenerate 500 days (times 8 or 9 indicators, times 100 FTSE stocks).

is there a way of just generating one days worth of values?

thanks in advance,
Nick James

**Comment by DaveSkender** [comment]
> is there a way of just generating one days worth of values?

Not yet, but we're working on that use case next.

Follow or continue this conversation in our [discussion on streaming indicators](https://github.com/DaveSkender/Stock.Indicators/discussions/1018) for progress and information about pre-release package versions to support live and incremental quote sources.  This is the next major feature area in our [roadmap](https://github.com/DaveSkender/Stock.Indicators/discussions/218)

**Comment by DaveSkender** [comment]
Nick, we agree with your sentiment on efficiency and are working on a more intentional design for incremental and streaming use cases.  In the meantime, re-creating a mostly redundant 500 period window is still quite inexpensive, [less than 0.1 milliseconds in most cases](https://dotnet.stockindicators.dev/performance/).
